echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java

```diff
@@ -59,34 +66,38 @@ public void handle(Event event, Map postedEvent) {
       return;
     }
 
-    GithubWebhookEvent githubWebhookEvent =
-        objectMapper.convertValue(postedEvent, GithubWebhookEvent.class);
+    GithubWebhookEvent webhookEvent = null;
+    String eventType = "";
 
-    event.content.put("hash", githubWebhookEvent.after);
-    if (githubWebhookEvent.ref == null) {
-      event.content.put("branch", "");
-    } else {
-      event.content.put("branch", githubWebhookEvent.ref.replace("refs/heads/", ""));
+    // TODO: Detect based on header rather than body key
```

Comment:
You should actually be able to `TODO` this as part of this PR. `event.details.requestHeaders` should have a copy of all the headers. You can probably do something like this:
```
String eventType = event.details.requestHeaders.getOrDefault("X-Github-Event", "")
```